src/Data/Singletons/Promote.hs

```diff
@@ -552,6 +554,7 @@ promotePat (DBangPa pat) = do
   qReportWarning "Strict pattern converted into regular pattern in promotion"
   (ty, pat') <- promotePat pat
   return (ty, DBangPa pat')
+promotePat (DSigPa _pat _ty) = error "TODO: Handle SigPa"
```

Comment:
To be honest, I'm not quite sure how one would promote a `DSigPa`. Suggestions welcome.